plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryTypeManager.java

```diff
@@ -380,6 +404,63 @@ private Optional<ColumnMapping> convertToTrinoType(Field field)
         }
     }
 
+    public List<ColumnMetadata> convertToTrinoType(List<String> names, List<String> types)
+    {
+        checkArgument(names.size() == types.size(), "Mismatched column names and types");
+
+        ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
+        for (int i = 0; i < names.size(); i++) {
+            TypeSignature typeSignature;
+            try {
+                TypeInfo typeInfo = getTypeInfoFromTypeString(types.get(i));
+                typeSignature = toTypeSignature(typeInfo);
+            }
+            catch (TrinoException | IllegalArgumentException e) {
+                // ignore unsupported types
+                continue;
+            }
+            columns.add(new ColumnMetadata(names.get(i), typeManager.getType(typeSignature)));
+        }
+        return columns.build();
+    }
+
+    private TypeSignature toTypeSignature(TypeInfo typeInfo)
+    {
+        return switch (typeInfo) {
+            case DecimalTypeInfo decimalTypeInfo:
+                yield createDecimalType(decimalTypeInfo.precision(), decimalTypeInfo.scale()).getTypeSignature();
+            case BigDecimalTypeInfo decimalTypeInfo:
+                yield createDecimalType(decimalTypeInfo.precision(), decimalTypeInfo.scale()).getTypeSignature();
+            case PrimitiveTypeInfo primitiveTypeInfo:
+                Type type = switch (primitiveTypeInfo.getStandardSqlTypeName()) {
+                    case BOOL -> BOOLEAN;
+                    case BYTES -> VARBINARY;
+                    case INT64 -> INTEGER;
+                    case FLOAT64 -> REAL;
+                    case DATE -> DATE;
+                    case DATETIME -> TIMESTAMP_MICROS;
+                    case STRING -> VARCHAR;
+                    case GEOGRAPHY -> VARCHAR;
+                    case JSON -> jsonType;
+                    case TIMESTAMP -> BIGINT;
```

Comment:
Why do we map to BIGINT?

Reply:
My mistake, I checked that we map this to `TIMESTAMP_TZ_MICRO`.